test/model/binding.dart

```diff
@@ -554,8 +554,32 @@ class FakeAndroidNotificationHostApi implements AndroidNotificationHostApi {
   }
   List<NotificationChannel> _createdChannels = [];
 
+  /// Consumes the log of calls made to [getNotificationChannels],
+  /// [deleteNotificationChannel] and [createNotificationChannel].
+  ///
+  /// Returns a list of function names in the order they were invoked.
+  List<String> takeChannelMethodCallLogs() {
+    final result = _channelMethodCallLogs;
+    _channelMethodCallLogs = [];
+    return result;
+  }
+  List<String> _channelMethodCallLogs = [];
+
+  @override
+  Future<List<NotificationChannel?>> getNotificationChannels() async {
+    _channelMethodCallLogs.add('getNotificationChannels');
+    return _createdChannels.toList(growable: false);
```

Comment:
I see, what the tests effectively do is call `createNotificationChannel` in order to simulate a channel having been created in the distant past.

That works fine. But this approach changes the semantics of `_createdChannels` — particularly because `deleteNotificationChannel` now removes channels from that list. Instead of a log of what was created, it's really serving now as a list of the channels that currently exist. The name, and the "take"-based API which is suited for a log, are no longer a fit.

So let's have a prep commit that replaces `takeCreatedChannels` with a getter like:
```dart
  Iterable<NotificationChannel> get createdChannels => _createdChannels;
```

Then that's well adapted for the direction it gets extended in the rest of this PR.